packages/backend/src/database/migrations/20250717082932_oauth_server_tokens.ts

```diff
@@ -0,0 +1,96 @@
+            expires_at TIMESTAMP NOT NULL,
+            redirect_uri TEXT NOT NULL,
+            scope TEXT[],
+            client_id VARCHAR(255) NOT NULL REFERENCES oauth2_clients(client_id) ON DELETE CASCADE,
+            user_id INTEGER NOT NULL REFERENCES users(user_id) ON DELETE CASCADE,
+            code_challenge VARCHAR(255),
+            code_challenge_method VARCHAR(10),
+            organization_uuid UUID NOT NULL REFERENCES organizations(organization_uuid) ON DELETE CASCADE,
+            created_at TIMESTAMP NOT NULL DEFAULT NOW()
+        );
+
+        CREATE TABLE IF NOT EXISTS oauth2_access_tokens (
+            access_token VARCHAR(255) PRIMARY KEY,
+            expires_at TIMESTAMP NOT NULL,
+            scope TEXT[],
+            client_id VARCHAR(255) NOT NULL REFERENCES oauth2_clients(client_id) ON DELETE CASCADE,
+            user_id INTEGER NOT NULL REFERENCES users(user_id) ON DELETE CASCADE,
+            organization_uuid UUID NOT NULL REFERENCES organizations(organization_uuid) ON DELETE CASCADE,
+            created_at TIMESTAMP NOT NULL DEFAULT NOW()
+        );
+
+        CREATE TABLE IF NOT EXISTS oauth2_refresh_tokens (
+            refresh_token VARCHAR(255) PRIMARY KEY,
+            expires_at TIMESTAMP NOT NULL,
+            scope TEXT[],
+            client_id VARCHAR(255) NOT NULL REFERENCES oauth2_clients(client_id) ON DELETE CASCADE,
+            user_id INTEGER NOT NULL REFERENCES users(user_id) ON DELETE CASCADE,
+            organization_uuid UUID NOT NULL REFERENCES organizations(organization_uuid) ON DELETE CASCADE,
+            created_at TIMESTAMP NOT NULL DEFAULT NOW()
+        );
+
+        CREATE INDEX IF NOT EXISTS oauth2_authorization_codes_client_id_idx ON oauth2_authorization_codes(client_id);
+        CREATE INDEX IF NOT EXISTS oauth2_authorization_codes_user_id_idx ON oauth2_authorization_codes(user_id);
+        CREATE INDEX IF NOT EXISTS oauth2_authorization_codes_expires_at_idx ON oauth2_authorization_codes(expires_at);
+        
+        CREATE INDEX IF NOT EXISTS oauth2_access_tokens_client_id_idx ON oauth2_access_tokens(client_id);
+        CREATE INDEX IF NOT EXISTS oauth2_access_tokens_user_id_idx ON oauth2_access_tokens(user_id);
+        CREATE INDEX IF NOT EXISTS oauth2_access_tokens_expires_at_idx ON oauth2_access_tokens(expires_at);
+        
+        CREATE INDEX IF NOT EXISTS oauth2_refresh_tokens_client_id_idx ON oauth2_refresh_tokens(client_id);
+        CREATE INDEX IF NOT EXISTS oauth2_refresh_tokens_user_id_idx ON oauth2_refresh_tokens(user_id);
+        CREATE INDEX IF NOT EXISTS oauth2_refresh_tokens_expires_at_idx ON oauth2_refresh_tokens(expires_at);
+
+        INSERT INTO oauth2_clients (client_id, client_secret, redirect_uris, grants, scopes, client_name) 
+        VALUES (
+            'lightdash-cli',
+            'secret',
+            ARRAY[${Array.from(
+                { length: 11 },
+                (_, i) => `'http://localhost:${8100 + i}/callback'`,
```

Comment:
whitelist ports between 8100 and 8110. Wildwards don't work

Reply:
which part of the stack doesn't support wild cards? In the oauth standard it should be allowed

Reply:
this is the code inside the library 

```
  async validateRedirectUri (redirectUri, client) {
    if (this.model.validateRedirectUri) {
      return this.model.validateRedirectUri(redirectUri, client);
    }

    return client.redirectUris.includes(redirectUri);
  }
  ```
  
  I'll add  a new method in the model to validate redirect uris taking wilcards into account